pkg/adapter/containers_remote.go

```diff
@@ -753,3 +776,59 @@ func (r *LocalRuntime) UnpauseContainers(ctx context.Context, cli *cliconfig.Unp
 	}
 	return ok, failures, nil
 }
+
+// Restart restarts a container over varlink
+func (r *LocalRuntime) Restart(ctx context.Context, c *cliconfig.RestartValues) ([]string, map[string]error, error) {
+	var (
+		containers        []*Container
+		restartContainers []*Container
+		err               error
+		ok                = []string{}
+		failures          = map[string]error{}
+	)
+	useTimeout := c.Flag("timeout").Changed || c.Flag("time").Changed
+	inputTimeout := c.Timeout
+
+	if c.Latest {
```

Comment:
This section looks pretty much the same as the section above, can you make it into a shared function?

Reply:
by above i assume you mean Restart in containers.go? If so, unfortunately no because we are calling out to different endpoints.  one remote and one libpod directly.

Reply:
@baude If we added filters to GetContainersByContext(), would that reduce some of the extra code across commands?

Reply:
the return types differ